### PR TITLE
fix some linter's complaints

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: controllers/idler/idler_controller.go [hunk 1]
```diff
@@ -132,6 +132,7 @@ func (r *Reconciler) ensureIdling(ctx context.Context, idler *toolchainv1alpha1.
 	}
 	newStatusPods := make([]toolchainv1alpha1.Pod, 0, 10)
 	for _, pod := range podList.Items {
+		pod := pod
```
Comment (alexeykazakov) [inline controllers/idler/idler_controller.go:135]: This is fixed in go 1.22 btw: https://go.dev/blog/loopvar-preview
So maybe worth adding a comment?
Something like:
```suggestion
		pod := pod // TODO We won't need it after upgrading to go 1.22: https://go.dev/blog/loopvar-preview
```

**Review by alexeykazakov — APPROVED** [review]

File: controllers/idler/idler_controller.go (hunk 1 above)
Comment (filariow) [reply controllers/idler/idler_controller.go]: done in 74e1f570dbc3d4dac759330185c0e764390d0ff1

File: controllers/nstemplateset/space_roles.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ func (r *spaceRolesManager) ensure(ctx context.Context, nsTmplSet *toolchainv1al
 	}
 	logger.Info("ensuring space roles", "namespace_count", len(nss), "role_count", len(nsTmplSet.Spec.SpaceRoles))
 	for _, ns := range nss {
+		ns := ns
```
Comment (mfrancisc) [inline controllers/nstemplateset/space_roles.go:34]:
```suggestion
		ns := ns // TODO We won't need it after upgrading to go 1.22: https://go.dev/blog/loopvar-preview
```

**Review by mfrancisc — APPROVED** [review]

File: controllers/nstemplateset/space_roles.go (hunk 2 above)
Comment (filariow) [reply controllers/nstemplateset/space_roles.go]: thanks! I somehow missed this line 😅

**Pull request merged**